### PR TITLE
Integrate 7/29 Nightly RN Build

### PR DESCRIPTION
**Pull request by NickGerleman**
Commits: https://github.com/facebook/react-native/compare/e96f1e1d8...949dedca9

(not adding a changelog for the description on this one since there are 120 commits)

Doing a larger integration than usual because web debugging was broken upstream from 7/9 to 7/28.

Not many changes that concern us apart from some JS dependency changes. Some of the changes pulled out into monorepo packages may effect us if changes happen to them that are unpublished, and we will need a better story for those soon.

We need to make local copies of flow typings that are currently in RN core but not published. Starting up a discussion there and tracking with #6139

###### Microsoft Reviewers: [Open in CodeFlow](http://wpcp.azurewebsites.net/CodeFlowProtocolProxy2.php?pullrequest=https://github.com/microsoft/react-native-windows/pull/6140)

### CODE REVIEW

File: packages/react-native-win32/flow-typed/npm/pretty-format_v26.x.x.js [hunk 1]
```diff
@@ -0,0 +1,51 @@
+/**
+ * (c) Facebook, Inc. and its affiliates. Confidential and proprietary.
+ *
+ * @flow strict
+ * @format
+ */
+
```
Comment (NickGerleman) [inline packages/react-native-win32/flow-typed/npm/pretty-format_v26.x.x.js:7]: This is synchronized with the RN Core OSS source tree. Guessing someone forgot to update the copyright headers when bringing this into OSS? FYI @TheSavior

**Review by rectified95 — APPROVED** [review]

File: packages/react-native-win32/src/Libraries/Utilities/BackHandler.win32.js [hunk 2]
```diff
@@ -0,0 +1,116 @@
...
+ * Detect hardware button presses for back navigation.
+ *
+ * Android: Detect hardware back button presses, and programmatically invoke the default back button
+ * functionality to exit the app if there are no listeners or if none of the listeners return true.
+ *
+ * tvOS: Detect presses of the menu button on the TV remote.  (Still to be implemented:
+ * programmatically disable menu button handling
+ * functionality to exit the app if there are no listeners or if none of the listeners return true.)
+ *
+ * iOS: Not applicable.
```
Comment (dannyvv) [inline packages/react-native-win32/src/Libraries/Utilities/BackHandler.win32.js:42]: Should we file a workitem to implement? Windows has support for 'back' button:
https://docs.microsoft.com/en-us/windows/win32/inputdev/virtual-key-codes
see: `VK_BROWSER_BACK`

File: vnext/flow-typed/npm/react-dom_v16.x.x.js [hunk 3]
```diff
@@ -0,0 +1,115 @@
+/**
```
Comment (dannyvv) [inline vnext/flow-typed/npm/react-dom_v16.x.x.js:1]: Just a generic comment. Shouldn't flow adopt the same pattern that TypeScript has by adding @flow-typed namespace to npm?

**Review by dannyvv — APPROVED** [review]

File: packages/react-native-win32/flow-typed/npm/pretty-format_v26.x.x.js (hunk 1 above)
Comment (TheSavior) [reply packages/react-native-win32/flow-typed/npm/pretty-format_v26.x.x.js:7]: Hmm, yeah, not sure why our license linter didn't force us to swap it. Thanks for flagging

File: packages/react-native-win32/src/Libraries/Utilities/BackHandler.win32.js (hunk 2 above)
Comment (NickGerleman) [reply packages/react-native-win32/src/Libraries/Utilities/BackHandler.win32.js:42]: We do have an implementation of this for the XAML RNW. We don't for react-native-win32/NetUI, but the built-in JavaScript behaves nicely when no native modules/events are present, so there isn't a reason to keep a forked stub of it.

File: vnext/flow-typed/npm/react-dom_v16.x.x.js (hunk 3 above)
Comment (NickGerleman) [reply vnext/flow-typed/npm/react-dom_v16.x.x.js:1]: I do think they have packages like that. Some of these differ from the public flow-typed packages though it looks like. So RN itself internally has its own typings stored in the file tree that its using that we want to replicate so we do type-checking of our sources in the same way.

**Pull request merged**